components/lib/fetch-okhttp/src/main/java/mozilla/components/lib/fetch/okhttp/OkHttpClient.kt

```diff
@@ -48,6 +58,10 @@ private fun okhttp3.OkHttpClient.rebuildFor(request: Request): okhttp3.OkHttpCli
             clientBuilder.followRedirects(false)
         }
 
+        if (request.cookiePolicy == Request.CookiePolicy.INCLUDE) {
+            clientBuilder.cookieJar(JavaNetCookieJar(cookieHandler))
+        }
```

Comment:
Is the Java cookie handler the only option here or could we also use an isolated cookie jar (not sure if okhttp comes with multiple implementations?)

Reply:
Nothing else built-in but we could provide our own implementation later if we want to e.g. if we want a persistent one (but it's tricky).